redis-hbo-provider/src/main/java/com/facebook/presto/statistic/HistoricalStatisticsSerde.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.statistic;
+
+import com.facebook.airlift.http.client.thrift.ThriftProtocolException;
+import com.facebook.airlift.http.client.thrift.ThriftProtocolUtils;
+import com.facebook.drift.codec.ThriftCodec;
+import com.facebook.drift.codec.ThriftCodecManager;
+import com.facebook.drift.transport.netty.codec.Protocol;
+import com.facebook.presto.spi.statistics.HistoricalPlanStatistics;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.SliceOutput;
+import io.airlift.slice.Slices;
+import io.lettuce.core.codec.RedisCodec;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Redis codec implementation for string keys and HistoricalPlanStatistics values.
+ */
+public class HistoricalStatisticsSerde
+        implements RedisCodec<String, HistoricalPlanStatistics>
+{
+    private static final int estimatedBuffersize = 100 * 1024;
+    private static final ThriftCodecManager thriftCodecManager = new ThriftCodecManager();
+
+    @Override
+    public String decodeKey(ByteBuffer bytes)
+    {
+        if (bytes.hasArray()) {
+            return StandardCharsets.UTF_8.decode(bytes).toString();
+        }
+        else {
+            throw new RuntimeException("Error decoding key");
+        }
+    }
+
+    @Override
+    public ByteBuffer encodeKey(String key)
+    {
+        return ByteBuffer.wrap(key.getBytes(StandardCharsets.UTF_8));
+    }
+
+    @Override
+    public ByteBuffer encodeValue(HistoricalPlanStatistics historicalPlanStatistics)
+    {
+        ThriftCodec<HistoricalPlanStatistics> writeCodec = thriftCodecManager.getCodec(HistoricalPlanStatistics.class);
```

Comment:
Extract this out into a member variable to cache?

Reply:
Is it possible to have this as a static member variable? Similar for the readCodec below

Reply:
I based this off [com.facebook.presto.hive.HiveMetadataUpdateHandleThriftSerde#thriftCodecManager](https://github.com/prestodb/presto/blob/7805bf0d451fe336afa77827e0697ab7ac2353ee/presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadataUpdateHandleThriftSerde.java#L50). Not sure if that needs to be initialized at each call but considered it safer to follow precedent